osquery/dispatcher/scheduler.h

```diff
@@ -43,6 +44,45 @@ class SchedulerRunner : public InternalRunnable {
   std::chrono::milliseconds getCurrentTimeDrift() const noexcept;
 
  private:
+   inline void calculateTimeDrift(std::chrono::time_point<std::chrono::steady_clock> start_time_point) {
```

Comment:
This actually sleeps so `calculateTimeDrift()` isn't the best name. Or you can refactor this method too to return "time to sleep" and then sleep outside this method.

Reply:
Oh I missed the sleep part.
Right, I'll edit the code.

Reply:
After staring at the code for a bit I think we can't extract `pauseMilli()` outside the function since the time drift variable is zeroed out after the thread sleeps.
I'm not sure if the order of execution matters.
I think I'll rename the method.